packages/components/src/palette-edit/index.tsx

```diff
@@ -261,32 +261,6 @@ function Option< T extends Color | Gradient >( {
 	);
 }
 
-/**
- * Checks if a color or gradient is a temporary element by testing against default values.
- */
-export function isTemporaryElement(
```

Comment:
I actually think this is a decent approach to removing the complexity. Why do we remove "default" values? To keep the list clean? We could rather trust the user to control how they want to create custom colors/gradients.

The following doesn't make much sense, but why not?

https://github.com/WordPress/gutenberg/assets/6458278/ccea7be7-0bd3-4371-81c4-55c9060685e5

The only thing we could probably address here is the lack of default value on the first item of the gradient:

<img width="521" alt="Screenshot 2023-12-21 at 11 03 18 am" src="https://github.com/WordPress/gutenberg/assets/6458278/1a9c19d6-e2ee-427e-aa82-1fddb59c35c2">

It has an empty value and can be saved.

Reply:
> The following doesn't make much sense, but why not?

That's a good point... maybe the user knows they're going to add a few custom colors and quickly goes to create them and then decides to go back later to update their colors? It might be nice to allow these sorts of interactions rather than try to predict too much, so I'm really liking the direction of this PR.